File: cp/LC2643-Row-With-Maximum-Ones.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define mod 1000000007

class Solution
{
public:
    vector<int> rowAndMaximumOnes(vector<vector<int>> &mat)
    {
        int ind = -1, cnt = -1;
        int i = 0;
        for (auto arr : mat)
        {
            int ct = count(arr.begin(), arr.end(), 1);
            if (ct > cnt)
            {
                cnt = ct;
                ind = i;
            }
            i += 1;
        }
        return {ind, cnt};
    }
};

int main()
{
    int n, m;
    cin >> n >> m;
    vector<vector<int>> mat(n, vector<int>(m));
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            cin >> mat[i][j];
        }
    }
    Solution sol;
    auto ans = sol.rowAndMaximumOnes(mat);
    cout << ans[0] << " " << ans[1] << endl;

    return 0;
}

File: cp/LC2643-Row-With-Maximum-Ones.py
from ast import List
class Solution:
    def rowAndMaximumOnes(self, mat: List[List[int]]) -> List[int]:
        ans=0
        for arr in mat:
            ans=max(ans,arr.count(1))
        return ans
if __name__=="__main__":
    n,m=map(int,input().split())
    mat=[]
    for _ in range(n):
        a=[int(x) for x in input().split()][:n]
        mat.append(a)
    sol=Solution()
    print(sol.rowAndMaximumOnes(mat))
    